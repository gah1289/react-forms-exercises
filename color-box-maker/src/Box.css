.delete-btn {
	position: relative;
	background-color: rgba(0, 0, 0, 0.5);
	color: rgba(255, 255, 255, 0.5);
	margin-right: 0;
	margin-left: auto;
	display: block;
	font-weight: bold;
	border: none;
}

.delete-btn:hover {
	background-color: rgba(0, 0, 0, 0.6);
	color: rgba(255, 255, 255, 1);
}

.hide-box {
	display: none;
}

.Box {
	margin: 10px;
	padding: 1px;
}
