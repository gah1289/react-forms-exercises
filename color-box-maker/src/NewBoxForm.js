import React, { useState } from 'react';

const NewBoxForm = ({ createBox }) => {
	const INITIAL_STATE = {
		width           : '',
		height          : '',
		backgroundColor : ''
	};

	const [
		formData,
		setFormData
	] = useState(INITIAL_STATE);

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((formData) => ({ ...formData, [name]: value }));
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		createBox({ ...formData });
		setFormData(INITIAL_STATE);
	};

	return (
		<form onSubmit={handleSubmit}>
			<label htmlFor="height">Box Height (px)</label>
			<input id="height" type="text" name="height" value={formData.height} onChange={handleChange} />
			<label htmlFor="width">Box Width (px)</label>
			<input id="width" type="text" name="width" value={formData.width} onChange={handleChange} />
			<label htmlFor="backgroundColor">Background Color</label>
			<input
				id="backgroundColor"
				type="text"
				name="backgroundColor"
				value={formData.backgroundColor}
				onChange={handleChange}
			/>
			<button>Add Box</button>
		</form>
	);
};

export default NewBoxForm;
