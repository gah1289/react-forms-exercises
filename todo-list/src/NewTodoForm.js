import React, { useState } from 'react';

const NewTodoForm = ({ createTask }) => {
	const INITIAL_STATE = {
		task : ''
	};

	const [
		formData,
		setFormData
	] = useState(INITIAL_STATE);

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((formData) => ({ ...formData, [name]: value }));
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		createTask({ ...formData });
		setFormData(INITIAL_STATE);
	};

	return (
		<form onSubmit={handleSubmit}>
			{' '}
			<label htmlFor="task">New Task: </label>
			<input
				id="task"
				type="text"
				name="task"
				value={formData.task}
				placeholder="Enter task here"
				onChange={handleChange}
			/>
			<button>Add Task</button>
		</form>
	);
};

export default NewTodoForm;
