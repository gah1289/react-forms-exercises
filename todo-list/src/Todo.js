import React, { useState } from 'react';
import './Todo.css';

const Todo = ({ task }) => {
	return <div className="Todo">{task}</div>;
};

export default Todo;
