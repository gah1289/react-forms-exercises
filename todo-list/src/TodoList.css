.remove-task {
	background-color: rgba(255, 0, 0, 0.5);
	border: none;
	border-radius: 100%;
	margin-left: 10px;
	color: white;
}

.remove-task:hover {
	background-color: rgba(255, 0, 0, 1);
}
