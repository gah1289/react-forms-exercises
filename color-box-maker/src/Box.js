import React, { useState } from 'react';

import './Box.css';

const Box = ({ backgroundColor, width, height }) => {
	const [
		hideBox,
		setHideBox
	] = useState(false);

	return (
		<div
			className={` Box ${hideBox ? 'hide-box' : ''}`}
			style={{ backgroundColor, width: Number(width), height: Number(height) }}
		>
			{' '}
			<button onClick={() => setHideBox(true)} className={`delete-btn`}>
				x
			</button>
		</div>
	);
};

export default Box;
