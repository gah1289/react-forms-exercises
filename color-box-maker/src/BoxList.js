import React, { useState } from 'react';
import NewBoxForm from './NewBoxForm';
import Box from './Box';
import { v4 as uuid } from 'uuid';

const BoxList = () => {
	const [
		boxes,
		setBoxes
	] = useState('');

	const createBox = (newBox) => {
		setBoxes((boxes) => [
			...boxes,
			{ newBox }
		]);
	};

	function showBoxes() {
		if (boxes.length > 0) {
			return (
				<div>
					{' '}
					{boxes.map((b) => (
						<Box
							key={uuid()}
							height={b.newBox.height}
							width={b.newBox.width}
							backgroundColor={b.newBox.backgroundColor}
						/>
					))}
				</div>
			);
		}
	}

	return (
		<div>
			<h1> Color Box Maker</h1>
			<NewBoxForm createBox={createBox} />
			{showBoxes()}
		</div>
	);
};

export default BoxList;
