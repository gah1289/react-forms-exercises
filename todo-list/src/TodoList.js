import React, { useState } from 'react';
import TodoForm from './NewTodoForm';
import Todo from './Todo';
import './TodoList.css';
import { v4 as uuid } from 'uuid';

const TodoList = () => {
	const [
		tasks,
		setTasks
	] = useState('');

	const createTask = (newTask) => {
		setTasks((tasks) => [
			...tasks,
			{ newTask }
		]);
	};

	function removeTask(task) {
		setTasks(tasks.filter((t) => t.newTask.task !== task));
	}

	function listTasksIfAny() {
		if (tasks.length > 0) {
			return (
				<div>
					<ol>
						{tasks.map((t, i) => (
							<li key={uuid()}>
								<Todo task={t.newTask.task} />
								<button className="remove-task" onClick={() => removeTask(t.newTask.task)}>
									x
								</button>
							</li>
						))}
					</ol>
				</div>
			);
		}
	}

	return (
		<div>
			{' '}
			<h1>To Do: </h1>
			{<TodoForm createTask={createTask} />}
			{listTasksIfAny()}
		</div>
	);
};

export default TodoList;
