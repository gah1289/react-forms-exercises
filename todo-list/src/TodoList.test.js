import React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';
import { remove } from '../../../es2015-module-exercise/src/helpers';

function addTask(todoList, task = 'Feed Hopper and Hank') {
	const taskInput = todoList.getByLabelText('New Task:');
	fireEvent.change(taskInput, { target: { value: task } });
	const button = todoList.getByText('Add Task');
	fireEvent.click(button);
}

function removeTask(todoList) {
	const removeButton = todoList.getByText('x');
	fireEvent.click(removeButton);
}

it('renders without crashing', function() {
	render(<TodoList />);
});

it('matches snapshot', function() {
	const { asFragment } = render(<TodoList />);
	expect(asFragment()).toMatchSnapshot();
});

it('can add a new item', function() {
	// no tasks on page yet
	expect(screen.queryByText('Feed Hopper and Hank')).not.toBeInTheDocument();
});

it('adds a task', function() {
	// task exists after adding it
	const todoList = render(<TodoList />);
	addTask(todoList);
	expect(screen.getByText('Feed Hopper and Hank')).toBeInTheDocument();
});

it('removes a task', function() {
	// task exists after adding it
	const todoList = render(<TodoList />);
	addTask(todoList);
	const task = screen.getByText('Feed Hopper and Hank');
	expect(task).toBeInTheDocument();
	removeTask(todoList);
	expect(task).not.toBeInTheDocument();
});
