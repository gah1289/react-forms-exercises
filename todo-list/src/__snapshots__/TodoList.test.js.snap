// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<DocumentFragment>
  <div>
     
    <h1>
      To Do: 
    </h1>
    <form>
       
      <label
        for="task"
      >
        New Task: 
      </label>
      <input
        id="task"
        name="task"
        placeholder="Enter task here"
        type="text"
        value=""
      />
      <button>
        Add Task
      </button>
    </form>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
     
    <h1>
      To Do: 
    </h1>
    <form>
       
      <label
        for="task"
      >
        New Task: 
      </label>
      <input
        id="task"
        name="task"
        placeholder="Enter task here"
        type="text"
        value=""
      />
      <button>
        Add Task
      </button>
    </form>
  </div>
</DocumentFragment>
`;
